sky/skylet/job_lib.py

```diff
@@ -406,16 +502,30 @@ def update_job_status(job_owner: str,
     # which contains the job status (str) and submission_id (str).
     job_detail_lists: List['ray_pydantic.JobDetails'] = job_client.list_jobs()
 
+    pending_jobs = _get_pending_jobs()
     job_details = {}
     ray_job_ids_set = set(ray_job_ids)
     for job_detail in job_detail_lists:
         if job_detail.submission_id in ray_job_ids_set:
             job_details[job_detail.submission_id] = job_detail
     job_statuses: List[Optional[JobStatus]] = [None] * len(ray_job_ids)
     for i, ray_job_id in enumerate(ray_job_ids):
+        job_id = job_ids[i]
         if ray_job_id in job_details:
             ray_status = job_details[ray_job_id].status
             job_statuses[i] = _RAY_TO_JOB_STATUS_MAP[ray_status]
+        if job_id in pending_jobs:
+            # Gives a 5 second timeout between job being submit from the
+            # pending queue until appearing in ray jobs
+            if pending_jobs[job_id]['submit'] > 0 and pending_jobs[job_id][
+                    'submit'] < time.time() - _PENDING_SUBMIT_TIMEOUT:
+                continue
```

Comment:
If we are using this now, we should get rid of the `submitted_gap_sec` in the following place, as that was for the same purpose for making sure the just submitted jobs are not marked as failed immediately.

https://github.com/skypilot-org/skypilot/blob/ab38a9366b2cd07a7b9b47ede514637a81cd05e9/sky/skylet/job_lib.py#L355